services/brig/src/Brig/API/Internal.hs

```diff
@@ -278,7 +282,9 @@ authAPI =
   Named @"legalhold-login" (callsFed (exposeAnnotations legalHoldLogin))
     :<|> Named @"sso-login" (callsFed (exposeAnnotations ssoLogin))
     :<|> Named @"login-code" getLoginCode
-    :<|> Named @"reauthenticate" reauthenticate
+    -- We qualify in place to avoid changing the internal API too much
+    -- FUTUREWORK?
+    :<|> Named @"reauthenticate" (\uid reauth -> qualifyLocal uid >>= \luid -> reauthenticate luid reauth)
```

Comment:
I don't understand this comment - can you clarify? 

Maybe we can also decide whether this should be `FUTUREWORK` or not

Reply:
i thought i fixed this and pushed the change, but it's gone. maybe an overly aggressive rebase?

anyway: cf7ec8b3f20dfdc8056d23127982e94af806247d